sdk/Pulumi/Deployment/Deployment.cs

```diff
@@ -169,6 +173,19 @@ internal Deployment(IEngine engine, IMonitor monitor, TestOptions? options)
 
         string IDeployment.OrganizationName => _organizationName;
         string IDeployment.ProjectName => _projectName;
+        string IDeployment.RootDirectory
+        {
+            get
+            {
+              if (_rootDirectory == null)
+              {
+                throw new InvalidOperationException("Root directory is not available in your version of the pulumi CLI, please upgrade to the latest version");
```

Comment:
```suggestion
                throw new InvalidOperationException("The Pulumi CLI version does not support root directory. Please update the Pulumi CLI.");
```
Standard phrasing copied from other similar errors.